2 - functional intro, data manipulation/andreea-ex1.js

```diff
@@ -0,0 +1,195 @@
+/* Data */
+const games = [
+    { name: 'alibaba', type: 'pacanele', jackpot: false },
+    { name: 'alibaba2', type: 'live', jackpot: false },
+    { name: 'gonzo cautare', type: 'poker', jackpot: false },
+    { name: 'alibaba', type: 'live', jackpot: true },
+    { name: 'dracula', type: 'pacanale', jackpot: false },
+    { name: 'dracula', type: 'pacanale', jackpot: true,
+        children: [
+            { name: 'cosmic fortune', type: 'pacanele', jackpot: true},
+            { name: 'gonzo cautare', type: 'poker', jackpot: false }
+        ]
+    }
+];
+
+/*EX01-flatten data*/
+function flatenedGames(data, nameProp){
+    var gamesFlatenedNames = data.reduce(
+        function (gamesName, current) {
+            var allName = gamesName.concat(current[nameProp]);
+            if (current.children) {
+                var children = current.children;
+                var childrenFlaten = children.reduce(
+                    function (gamesChialdName, currentChiald) {
+                        var newChildren = gamesChialdName.concat(currentChiald[nameProp]);
+                        return newChildren;
+                    }, []);
+                var allGameName = allName.concat(childrenFlaten);
+                return allGameName;
+            }
+            return allName;
+        }
+        , []);
+    return gamesFlatenedNames;
+}
+
+//*Returne all positive jackpot names
+function boolProp(data, bool ) {
+    var stockAllTrueJAckpot = [];
+    for (var prop in data) {
+        if (data[prop]['jackpot'] == bool ){
```

Comment:
always use '===' in javascript for type coercion and for speed. The correct check in this case would be typeof data...===boolean